src/OpenLoco/src/Localisation/Formatting.cpp

```diff
@@ -78,56 +78,65 @@ namespace OpenLoco::StringManager
         return kMonthToStringMap.find(month)->second;
     }
 
-    static char* formatInt32Grouped(int32_t value, char* buffer)
+    class ThousandsSepFacet : public std::numpunct<char>
     {
-        registers regs;
-        regs.eax = (uint32_t)value;
-        regs.edi = X86Pointer(buffer);
+        char _sep;
+
+    public:
+        explicit ThousandsSepFacet(char sep)
+            : _sep(sep)
+        {
+        }
+
+    protected:
+        [[nodiscard]] char do_thousands_sep() const override
+        {
+            return _sep;
+        }
 
-        call(0x00495F35, regs);
-        return X86Pointer<char>(regs.edi);
+        [[nodiscard]] std::string do_grouping() const override { return "\3"; }
+    };
+
+    // 0x00495F35
+    static char* formatInt32Grouped(int32_t value, char* buffer)
+    {
+        auto* ret = fmt::format_to(buffer, std::locale(std::locale(), new ThousandsSepFacet(',')), "{:L}", value);
+        *ret = '\0';
+        return ret;
     }
 
+    // 0x00495E2A
     static char* formatInt32Ungrouped(int32_t value, char* buffer)
     {
-        registers regs;
-        regs.eax = (uint32_t)value;
-        regs.edi = X86Pointer(buffer);
-
-        call(0x495E2A, regs);
-        return X86Pointer<char>(regs.edi);
+        auto* ret = fmt::format_to(buffer, "{}", value);
+        *ret = '\0';
+        return ret;
     }
 
+    // 0x00496052
     static char* formatInt48Grouped(uint64_t value, char* buffer, uint8_t separator)
     {
-        registers regs;
-        regs.eax = (uint32_t)value;
-        regs.edx = (uint32_t)(value / (1ULL << 32)); // regs.dx = (uint16_t)(value >> 32);
-        regs.edi = X86Pointer(buffer);
-        regs.ebx = (uint32_t)separator;
-
-        call(0x496052, regs);
-        return X86Pointer<char>(regs.edi);
+        auto* ret = fmt::format_to(buffer, std::locale(std::locale(), new ThousandsSepFacet(',')), "{:L}", value * static_cast<uint64_t>(std::pow(10, separator)));
```

Comment:
What is the last parameter meant to do?

Reply:
you mean the `value * static_cast<uint64_t>(std::pow(10, separator))`? The separator is the number of 0's to add onto the value. So a separator of 0 is 10^0 == 1 == No zeros. If 1 its 10^1 == 10 == 1 zero.

Reply:
Okay, so the `separator` bit is a misnomer, then. Perhaps we should call it a decimalisationFactor or something?

Still, curious how this is used. I suppose the 48 bit integer format is the only one used exclusively for currency. Either way, I noticed the value is set to 0 for Dollars and Pounds. Suppose it's 2 for Yen, for example?